parts/kubernetesagentcustomdata.yml

```diff
@@ -128,7 +128,8 @@ write_files:
   owner: "root"
   content: |
     #!/bin/bash
-    exit 0
+    echo $(curl -q http://169.254.169.254/metadata/v1/InstanceInfo 2>/dev/null | sed "s/.*\"FD\":\"\([0-9]*\).*/\1/") > /etc/kubernetes/fd
```

Comment:
1. What the url (http://169.254.169.254/metadata/v1/InstanceInfo) for?
2. This url contain explicit IP address, and the link itself is not very meaningful. So is it possible to find an alternative link?
2. For curl, i think we should do retry.